src/lib/list/selection-list.ts

```diff
@@ -374,12 +374,16 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
 
   /** Selects all of the options. */
   selectAll() {
-    this._setAllOptionsSelected(true);
+    if (!this._disabled) {
```

Comment:
Rather than checking whether the list is disabled, these should loop through the options and check whether each individual one is disabled. That would catch both the case where individual ones are disabled and the entire list is disabled.

Reply:
Yes, that indeed sounds like a better solution.